### PR TITLE
ARM Buyback

### PR DESCRIPTION
**Pull request by clement-ux**
# Create a new buyback for ARM

Create a new buyback for ARM, using `WETH` as `oToken`. This use the same contract as `OETHBuyback.sol` and `OUSDBuyback.sol`.

# Params
Using the same parameter at OETHBuyback.sol except for oToken.
- Otoken : [WETH](https://etherscan.io/address/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2)
- SwapRouter: [Swapper1InchV5](https://etherscan.io/address/0xcD0fcF8a31Bc78ec07752e9CCD3960E936D18366)
- Strategist: [Origin MS 2](https://etherscan.io/address/0xF14BBdf064E3F67f51cd9BD646aE3716aD938FDC)
- TreasuryManager: [Origin MS 2](https://etherscan.io/address/0xF14BBdf064E3F67f51cd9BD646aE3716aD938FDC)
- RewardsSource: [OGN Rewards source ](https://etherscan.io/address/0x7609c88E5880e934dd3A75bCFef44E31b1Badb8b)
- Proxy Governor: [Timelock](https://etherscan.io/address/0x35918cDE7233F2dD33fA41ae3Cb6aE0e42E0e69F#code)

## Code Change Checklist

To be completed before internal review begins:

- [x]  The contract code is complete
  - No smart contract changes.
- [x]  Executable deployment file
- [x]  Fork tests that test after the deployment file runs
- [x]  Unit tests *if needed
- [x]  The owner has done a [full checklist review](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) of the code + tests

Internal review:

- [ ] Two approvals by internal reviewers

## Deploy checklist

Two reviewers complete the following checklist:

```
- [ ] All deployed contracts are listed in the deploy PR's description
- [ ] Deployed contract's verified code (and all dependencies) match the code in master
- [ ] The transactions that interacted with the newly deployed contract match the deploy script.
- [ ] Governance proposal matches the deploy script
- [ ] Smoke tests pass after fork test execution of the governance proposal
```

### CODE REVIEW

File: contracts/deploy/mainnet/107_arm_buyback.js.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      "ARMBuyback",
+      dARMBuybackProxy.address
+    );
+
+    // 3. Construct initialize call data to initialize and configure the new buyback
+    const initData = cARMBuyback.interface.encodeFunctionData(
+      "initialize(address,address,address,address,uint256)",
+      [
+        cSwapper.address, // Swapper1InchV5
+        "0xF14BBdf064E3F67f51cd9BD646aE3716aD938FDC", // MS 2
```
Comment (DanielVF) [inline contracts/deploy/mainnet/107_arm_buyback.js.js:41]: Let's look these up from `addresses.mainnet`, instead of having them separately coded here.

**Review by DanielVF — DISMISSED** [review]

**Comment by DanielVF** [comment]
@clement-ux Can you make the linter happy? Thanks.

**Comment by clement-ux** [comment]
> @clement-ux Can you make the linter happy? Thanks.

I was sure I'd done it before push. Sorry, fixed now.

File: contracts/deploy/mainnet/107_arm_buyback.js.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "ARMBuyback",
+      dARMBuybackProxy.address
+    );
+
+    // 3. Construct initialize call data to initialize and configure the new buyback
+    const initData = cARMBuyback.interface.encodeFunctionData(
+      "initialize(address,address,address,address,uint256)",
+      [
+        cSwapper.address, // Swapper1InchV5
+        addresses.mainnet.MultiSig2OutOf9, // MS 2
```
Comment (naddison36) [inline contracts/deploy/mainnet/107_arm_buyback.js.js:41]: you can get the strategist address with
```js
const { strategistAddr } = await getNamedAccounts();
```

This is configured in the Hardhat config

**Review by naddison36 — CHANGES_REQUESTED** [review]
Looks good. My only change would be getting the strategist address

**Review by naddison36 — APPROVED** [review]

**Pull request merged**